pkgs/development/ocaml-modules/tls/eio.nix

```
@@ -0,0 +1,35 @@
+{
+  buildDunePackage,
+  crowbar,
+  eio,
+  eio_main,
+  mdx,
+  mirage-crypto-rng-eio,
+  ptime,
+  tls,
+}:
+
+buildDunePackage rec {
+  pname = "tls-eio";
+
+  inherit (tls) src meta version;
+
+  minimalOCamlVersion = "5.0";
+
+  doCheck = true;
+  nativeCheckInputs = [
+    mdx.bin
+  ];
+  checkInputs = [
+    crowbar
+    eio_main
+    mdx
```

```suggestion
    (mdx.override { inherit logs; })
```

Instead of the (controversial) first commit, you can simply locally override the `mdx` library to use the version of `logs` of your liking.